__tests__/util/request-manager.js

```diff
@@ -203,13 +204,34 @@ for (const statusCode of [403, 442]) {
       resolve: body => {},
       reject: err => {
         expect(err.message).toBe(
-          `https://localhost:port/?nocache: Request "https://localhost:port/?nocache" returned a 403`,
+          `https://localhost:port/?nocache: Request "https://localhost:port/?nocache" returned a ${statusCode}`,
         );
       },
     });
   });
 }
 
+test('RequestManager.execute propagates non HTTP errors', async () => {
+  jest.resetModules();
+  jest.mock('request', factory => options => {
+    options.callback(new Error('bad stuff happened'), {}, '');
+    return {
+      on: () => {},
+    };
+  });
+  const config = await await Config.create({}, new Reporter());
+  await config.requestManager.execute({
+    params: {
+      url: `https://localhost:port/?nocache`,
+      headers: {Connection: 'close'},
+    },
+    resolve: body => {},
+    reject: err => {
```

Comment:
Shouldn't this callback be spied on? At the moment Jest doesn't check whether it's actually called 🤔

Reply:
I was actually just following existing convention, but it looks like the tests are [invoking](https://github.com/yarnpkg/yarn/pull/7000/files#diff-ae8e937424abd07f0db1a5b3f6532758R217) the [callback](https://github.com/yarnpkg/yarn/blob/dac22ddfea657946f5c2943cab69bff7dae63ebc/src/util/request-manager.js#L412) manually, which runs the "async" resolution. The `await` in front of `config.requestManager.execute` is misleading because RequestManager#execute is a void method.

The tests technically test the correct behavior, but could probably be a bit more straightforward.